Sources/NIOConcurrencyHelpers/NIOAtomic.swift

```diff
@@ -223,6 +223,7 @@ public final class NIOAtomic<T: NIOAtomicPrimitive> {
     /// - Returns: `True` if the exchange occurred, or `False` if `expected` did not
     ///     match the current value and so no exchange occurred.
     @inlinable
+    @discardableResult
```

Comment:
this shouldn't be discardable. If people don't care about whether this succeeds, they should use `store`